src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/services/TaskManagementService.java

```diff
@@ -374,6 +395,20 @@ public void cancelTask(String taskId,
         }
     }
 
+    private boolean checkUserHasCancelPermission(List<RoleAssignment> roleAssignments,
+                                                   Set<TaskRoleResource> taskRoleResources) {
+        for (RoleAssignment roleAssignment: roleAssignments) {
```

Comment:
Do you need to loop through RoleAssignment list? TaskResource will returned with TaskRoleResource of only current user. So if user has got cancel permission for any of his role this will return true regradless of you check against role name.

Reply:
Logic rewritten to avoid role name check